mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -418,25 +416,14 @@ void AbstractControllerExecution::run()
         condition_.notify_all();
       }
 
-      boost::chrono::thread_clock::time_point end_time = boost::chrono::thread_clock::now();
-      boost::chrono::microseconds execution_duration =
-          boost::chrono::duration_cast<boost::chrono::microseconds>(end_time - loop_start_time);
-      configuration_mutex_.lock();
-      boost::chrono::microseconds sleep_time = calling_duration_ - execution_duration;
-      configuration_mutex_.unlock();
       if (moving_ && ros::ok())
       {
-        if (sleep_time > boost::chrono::microseconds(0))
-        {
-          // interruption point
-          boost::this_thread::sleep_for(sleep_time);
-        }
-        else
+        // provide an interruption point also with 0 or negative sleep_time
```

Comment:
the comment should be now something like "provide an interruption point, as the nanosleep used by ROS Timer is not  interruptable"

Reply:
Would it make sense to add another interruption_point after the sleep?  :thinking:

Reply:
Maybe at the start of the loop makes more sense? On case one interrupts immediately

Reply:
Although before and after sleep might be more explicit because that's the actual reason

Reply:
yap,,,  taht's what I thought;  but looks a bit... shoddy, hehe.   I'm good with either option;   will give a try